pre_build/cccp_index_pre_build_check.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+"""Create jenkins jobs from the container-index."""
+
+import os
+import subprocess
+import sys
+import tempfile
+from glob import glob
+
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+jjb_defaults_file = 'pre-build-job.yml'
+
+required_attrs = ['image_name', 'image_version']
+optional_attrs = ['rundotshargs']
+overwritten_attrs = ['jobid', 'git_url', 'appid', 'jobs']
+
+
+def projectify(
+        new_project, appid, jobid, giturl, gitpath, gitbranch,
+        desiredtag, prebuild_script):
+
+    new_project[0]['project']['appid'] = appid
+    new_project[0]['project']['jobid'] = jobid
+    new_project[0]['project']['name'] = appid
+    new_project[0]['project']['git_url'] = giturl
+    new_project[0]['project']['git_branch'] = gitbranch
+    rel_path = "/"
+    if gitpath and gitpath != "/":
+        rel_path = gitpath if gitpath.startswith("/") else (rel_path + gitpath)
+    new_project[0]['project']['rel_path'] = rel_path
+    new_project[0]['project']['jobs'] = ['cccp-rundotsh-job']
+
+    if 'rundotshargs' not in new_project[0]:
+        new_project[0]['project']['rundotshargs'] = ''
+    elif new_project[0]['project']['rundotshargs'] is None:
+        new_project[0]['project']['rundotshargs'] = ''
+
+    new_project[0]['project']['desired_tag'] = desiredtag
+    new_project[0]['project']['prebuild_script'] = prebuild_script
+    return new_project
+
+
+def get_projects_from_index(indexdlocation):
+    projects = []
+    for yamlfile in glob(indexdlocation + "/*.yml"):
+        if "index_template" not in yamlfile:
+            stream = open(yamlfile, 'r')
+            index_yaml = yaml.load(stream, Loader=yaml.BaseLoader)
+            stream.close()
+            # print index_yaml['Projects']
+
+            for project in index_yaml['Projects']:
+                if(project['git-url'] is None):
+                    continue
+                else:
+                    try:
+                        appid = project['app-id']
+                        jobid = project['job-id']
+                        giturl = project['git-url']
+                        gitpath = project['git-path'] \
+                            if (project['git-path'] is not None) else ''
+                        gitbranch = project['git-branch']
+                        try:
+                            desiredtag = project['desired-tag'] \
+                                if (project['desired-tag'] is not None) \
+                                else 'latest'
+                        except Exception:
+                            desiredtag = 'latest'
```

Comment:
`try..except` seems a bit over-engineering to me here. How about?:
```python
In [2]: project = dict()

In [4]: desiredtag = project.get('desired-tag') if (project.get('desired-tag') is not None) else 'latest'

In [5]: desiredtag
Out[5]: 'latest'

In [6]: project['desired-tag'] = 'old'

In [7]: desiredtag = project.get('desired-tag') if (project.get('desired-tag') is not None) else 'latest'

In [8]: desiredtag
Out[8]: 'old'
```

Reply:
hmm, its an old code did not change. done changes.